### ISSUE TITLE
`IllegalArgumentException` when using `AccessibilityRenderExtension` with `SHRINK`

### ISSUE THREAD

**Issue by VirtualParticle** [cashapp/paparazzi#1684]
**Description**
When running a test with the `AccessibilityRenderExtension` and the `SHRINK` rendering mode, the test crashes with an `IllegalArgumentException`

**Steps to Reproduce**

This test produces the error:

```kotlin
@get:Rule
val paparazzi = Paparazzi(
  renderExtensions = setOf(AccessibilityRenderExtension()),
  renderingMode = SessionParams.RenderingMode.SHRINK,
)

@Test
fun example() {
  paparazzi.snapshot {
    Text("Hello, world!")
  }
}
```
```
SEVERE: broken: View draw failed
java.lang.IllegalArgumentException: Layout: -84 < 0
	at android.text.Layout.<init>(Layout.java:304)
	at android.text.StaticLayout.<init>(StaticLayout.java:634)
	at android.text.StaticLayout$Builder.build(StaticLayout.java:444)
	at app.cash.paparazzi.accessibility.AccessibilityOverlayDetailsView.draw(AccessibilityOverlayDetailsView.kt:80)
```

The error occurs in `AccessibilityOverlayDetailsView`:

```kotlin
accessibilityElements.forEach {
  paint.color = it.color.toColorInt()
  val badge = RectF(margin, lastYCoord, margin + rectSize, lastYCoord + rectSize)
  canvas.drawRoundRect(badge, cornerRadius, cornerRadius, paint)
  canvas.save()

  val text = it.contentDescription
  val textX = badge.right + innerMargin
  val textY = badge.top
  val textLayoutWidth = (width - textX).toInt()

  val textLayout = StaticLayout.Builder
    .obtain(text, 0, text.length, textPaint, textLayoutWidth)
    .setEllipsize(TextUtils.TruncateAt.END)
    .build() // java.lang.IllegalArgumentException: Layout: -84 < 0
  canvas.save()
```

**Expected behavior**
The test should execute normally.

**Additional information:**
- Paparazzi Version: 1.3.5
- OS: MacOS and Linux
- Compile SDK: 34
- Gradle Version: 8.10.2
- Android Gradle Plugin Version: 8.4.2

**Comment by savvisingh** [comment]
I have been facing the same issue, looks like there were changes in invalidPrintableChars in Snapshot class which makes some characters invalid in test name, In my case all my tests contain the invalid character, Would really like a workaround.

**Comment by jrodbx** [comment]
@savvisingh aside from the commonality in throwing an instance of IllegalArgumentException, it doesn't appear that the issue you're describing is related.  The breaking change is intended to avoid invalid chars from crashing on Windows platforms (https://github.com/cashapp/paparazzi/issues/1532).  Accordingly, please ensure your tests sanitize the labels and/or test method arguments.

**Comment by savvisingh** [comment]
Ah true i didn't read the whole error message before commenting, haha thanks for the reply anyway

**Comment by colinmarsch** [comment]
Closing this issue as a duplicate of https://github.com/cashapp/paparazzi/issues/1350. As well, in that issue [this PR](https://github.com/cashapp/paparazzi/pull/1795) is linked which changes the behaviour here to log an error message if `RenderingMode.SHRINK` is used with the `AccessibilityRenderExtension`.
